Adopt delegation NEFilterSource SPI

pvollan opened the pull request:
#### 4b893edcda1fb9040803a678003359ce823f7a2f
<pre>
Adopt delegation NEFilterSource SPI
<a href="https://bugs.webkit.org/show_bug.cgi?id=258304">https://bugs.webkit.org/show_bug.cgi?id=258304</a>
rdar://109900099

Reviewed by Brent Fulgham.

Adopt delegation NEFilterSource SPI, providing audit token of the UI process.

* Source/WebCore/PAL/pal/spi/cocoa/NEFilterSourceSPI.h:
* Source/WebKit/Configurations/WebKit.xcconfig:
* Source/WebKit/NetworkProcess/cocoa/NetworkProcessCocoa.mm:
(WebKit::NetworkProcess::platformInitializeNetworkProcessCocoa):

Canonical link: <a href="https://commits.webkit.org/265463@main">https://commits.webkit.org/265463@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/22fc03bf9484bcc9487ccac0a51d9117abd556fb

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/10952 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/11165 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/11483 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/12599 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/10461 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/10968 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/13542 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/11147 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/13391 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/11111 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/15/builds/12011 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/9234 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/13005 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/9304 "Passed tests") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/10/builds/9897 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/17123 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/10375 "Passed tests") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/36/builds/10050 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/13284 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/10487 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/17/builds/8578 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/9661 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/2632 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/13928 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/10343 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

brentfulgham commented on Source/WebKit/NetworkProcess/cocoa/NetworkProcessCocoa.mm:
```diff
@@ -98,6 +102,12 @@ static void initializeNetworkSettings()
     [NSURLCache setSharedURLCache:urlCache.get()];
+
+#if ENABLE(CONTENT_FILTERING_IN_NETWORKING_PROCESS)
+    auto auditToken = *parentProcessConnection()->getAuditToken();
```
Is it safe to dereference this `optional<>` here? If the audit token is not available for some reason (perhaps older OS?) this produces an undefined return value.

pvollan replied:
Ah, good point, fixed in latest patch!

Thanks for reviewing!

brentfulgham reviewed (approved):
Looks okay, but I'm concerned about the dereference of the optional. Can you double check that this is correct? We check for null opt in other cases.

pvollan committed:
Adopt delegation NEFilterSource SPI
https://bugs.webkit.org/show_bug.cgi?id=258304
rdar://109900099

Reviewed by Brent Fulgham.

Adopt delegation NEFilterSource SPI, providing audit token of the UI process.

* Source/WebCore/PAL/pal/spi/cocoa/NEFilterSourceSPI.h:
* Source/WebKit/Configurations/WebKit.xcconfig:
* Source/WebKit/NetworkProcess/cocoa/NetworkProcessCocoa.mm:
(WebKit::NetworkProcess::platformInitializeNetworkProcessCocoa):

Canonical link: https://commits.webkit.org/265463@main
